### PR TITLE
conditionNotMet is not a request error

### PR DESCRIPTION
**Pull request by olaaun**
Ref [#23075](https://github.com/vespa-engine/vespa/issues/23075)

### CODE REVIEW

**Review by jonmv — APPROVED** [review]

**Comment by jonmv** [comment]
412 is also not the right HTTP status code for this, imho 😝

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java [hunk 1]
```diff
@@ -1407,11 +1407,13 @@ private MeasuringResponseHandler(ResponseHandler delegate, com.yahoo.documentapi
 
         @Override
         public ContentChannel handleResponse(Response response) {
-            switch (response.getStatus() / 100) {
-                case 2: metrics.reportSuccessful(type, start); break;
-                case 4: metrics.reportFailure(type, DocumentOperationStatus.REQUEST_ERROR); break;
-                case 5: metrics.reportFailure(type, DocumentOperationStatus.SERVER_ERROR); break;
-            }
+            var statusCodeGroup = response.getStatus() / 100;
+            if (statusCodeGroup == 2 || response.getStatus() == 412)
```
Comment (gjoranv) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:1411]: A comment seems appropriate here. 🙂

Comment (bratseth) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:1411]: agreed

Comment (olaaun) [reply vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:1411]: https://github.com/vespa-engine/vespa/pull/23120

**Pull request merged**